src/utilities/hypre_hopscotch_hash.h

```diff
@@ -615,10 +630,12 @@ static inline HYPRE_Int hypre_UnorderedIntSetContains( hypre_UnorderedIntSet *s,
     if (hash == s->hash[currElm] && key == s->key[currElm])
       return 1;
     hopInfo &= ~(1U << i);
-  } 
+  }
 
+#ifdef HYPRE_CONCURRENT_HOPSCOTCH
   if (segment->timestamp == startTimestamp)
```

Comment:
This is not related to this PR. Looking at the original code, it is mysterious for me. How can the if-statement on line 620 is not true if we set it equal in line 609. I don't understand the stuff in between can break the equality. Then, what's the point of having stuff after line 623? Maybe I missed a lot things here. Again, I haven't understood how HOPSCOTCH works yet.

Reply:
I agree with you, and the stuff after line 623 could be deleted. However, I decided to leave as it is since it doesn't affect the purpose of this PR.

Reply:
OK. Maybe this is fine. It is actually the lines after 623 that bother me. Originally the function is guaranteed to `return 0` at line 621, since it is supposed to be with `#ifdef HYPRE_CONCURRENT_HOPSCOTCH` only. But now when without `#ifdef HYPRE_CONCURRENT_HOPSCOTCH`, it proceeds to the lines after. Just from a high level, the function seems to have different behavior.

Reply:
I understand what you are saying now. Previously those lines could never be reached but now this might be possible. Should I exclude them, or maybe put them inside an `#if 0 ... #endif ` region?

Reply:
@victorapm No, I think we may need it for the non-`HYPRE_CONCURRENT_HOPSCOTCH` case. It appears like a standard loop searching for hash values and the key. It is safer to leave it here. At least we can guarantee, with `HYPRE_CONCURRENT_HOPSCOTCH` the code is exactly as before. It doesn't seem this function is used anywhere in Hypre. Is this used in AMG-DD maybe? 

We need to remind ourselves to pay attention when using these functions without `HYPRE_CONCURRENT_HOPSCOTCH`. It doesn't seem we have tested them before. 

My 2 cents

Reply:
Thanks, @liruipeng! That makes sense. This function is not being used right now, however, its BigInt counterpart (which has the same logic) is being used in `aux_interp.c`